src/equivalent.js

```diff
@@ -0,0 +1,59 @@
+import { isObject, isDate, isArray } from './types.js';
+
+/**
+ * Recursive objects equivalence check.
+ * @param {*} obj1 The original object.
+ * @param {*} obj2 The object to compare
+ * @param {Array} [processing] A list of already processed comparisons.
+ * @return {Boolean}
+ */
+export default function equivalent(obj1, obj2, processing = []) {
```

Comment:
I usually don't like exposing an "internal" function parameter like `processing` to the userland. Instead, I generally prefer a "cleaner" signature that internally uses a private worker:

```js
function equivalentWorker (obj1, obj2, processing) {
   /* ... */
}

export default function equivalent(obj1, obj2) {
    return equivalentWorker(obj1, obj2, []);
}
```